### ISSUE TITLE
Resources fail to load, when defined across multiple files, on unbuntu

### ISSUE THREAD

**Issue by flah00** [ruby-grape/grape#379]
**TL;DR** Resources defined in more than one file behave differently on different operating systems. If I define a nested resource, based on a has_many relationship, in separate files, I see the expected results on Mac OS X, but I don't see the expected results on Ubuntu.

We're using FDD, as suggested in http://code.dblock.org/modularizing-a-ror-grape-api-multiple-versions. I would expect grape to serve the same routes, whether on Ubuntu or Mac OS X. I find that when I run the following code on my Mac OS X box, the routes are correctly served... However, when I push this up to Heroku, my routes return 404 as often as they return a valid response.

``` ruby
# config.ru
require ::File.expand_path('../config/environment',  __FILE__)
require 'rack/cors'

api = Rack::Builder.new do
  use Rack::CommonLogger
  use Rack::Cors do
    allow do
      origins '*'
      resource '*', headers: :any, methods: [:delete, :get, :post, :put]
    end
  end
  run Api
end

run Rack::URLMap.new \
  '/'     => Ours::Application,
  '/api'  => api
```

``` ruby
# app/api/api.rb
class Api < Grape::API
  version 'v1'
  logger Rails.logger

  default_format :json
  formatter :csv, lambda { |object,e| Ours::Api.tsv_formatter(object, :csv) }
  formatter :xls, lambda { |object,e| Ours::Api.tsv_formatter(object, :xls) }

  content_type Mime::JSON.to_sym, Mime::JSON.to_s
  content_type Mime::CSV.to_sym, Mime::CSV.to_s
  content_type Mime::XLS.to_sym, Mime::XLS.to_s

  http_basic do |username, password|
    [username, password] == ['blah', 'bl4h']
  end

  rescue_from ActiveRecord::RecordNotFound do |e|
    class_eval {include ::Api.helpers}
    id = log_error(e)
    rack_response({error: "The document you requested could not be found", id: id}.to_json, 404)
  end

  rescue_from ActiveRecord::RecordInvalid do |e|
    class_eval {include ::Api.helpers}
    id = log_error(e)
    rack_response({error: e.to_s, id: id}.to_json, 400)
  end

  rescue_from :all do |e|
    class_eval {include ::Api.helpers}
    if Rails.env.development?
      raise e
    else
      id = log_error(e)
      rack_response({error: e.class.name, id: id}.to_json, 500)
    end
  end

  helpers do
    def log_error(e)
      id = env['api.endpoint'].inspect.hash.abs
      ::Api.logger.error(<<-EOT)
        CLASS:#{e.class} MESSAGE:[#{e}]
        URI:#{env['REQUEST_URI']}
        METHOD:#{env['REQUEST_METHOD']}
        CLIENT:#{env['REMOTE_ADDR']}
        API: #{env['api.format']}
        PAYLOAD: #{env['rack.request.form_hash'].inspect}
        ID: #{id}
      EOT
      id
    end
  end

  before do
    Ours::Cube.send_env "campaign_group_api_request", env, cookies: @cookies.read(@request)

    if %w[xls csv].include? params['format']
      header['Content-Disposition'] = "attachment; filename=report.#{params['format']}"
    end
  end

  resource '/' do
    get '/version' do
      {version: env['api.version']}
    end
  end

  Dir["#{Rails.root}/app/api/api_*/**/*.rb"].each do |file|
    obj = file.sub(/.*(api_.*).rb/, '\1').camelize
    include obj.constantize
  end
end
```

``` ruby
# app/api/api_v1/foos.rb
module ApiV1::Foos
  extend ActiveSupport::Concern
  included do
    resource :foos do
      get '/' do
        Foo.all
      end
      resource "/:id" do
        get '/' do
          Foo.find(params[:id])
        end
      end
    end
  end
end
```

``` ruby
# app/api/api_v1/foos/bars.rb
module ApiV1::Foos::Bars
  extend ActiveSupport::Concern
  included do
    resource 'foos/:id' do
      resource :bars do
        get '/' do
          Bar.where(foo_id: params[:id])
        end
      end
    end
  end
end
```
- Routes (they are the same on heroku and local)

```
> pp Api.routes
[version=v1, method=GET, path=/:version/foos(.:format),
 version=v1, method=GET, path=/:version/foos/:id(.:format),
 version=v1, method=GET, path=/:version/foos/:id/bars(.:format)]
```

```
$ bundle list
Gems included by the bundle:
  * actionmailer (3.2.6)
  * actionpack (3.2.6)
  * activemodel (3.2.6)
  * activerecord (3.2.6)
  * activerecord-has_money (0.0.1 5cc3d22)
  * activeresource (3.2.6)
  * activesupport (3.2.6)
  * ours-apis (0.0.1 62529d7)
  * ours-cube (0.1.1 878f485)
  * ours-resque (0.0.4 f614674)
  * addressable (2.3.2)
  * airbrake (3.1.9)
  * arel (3.0.2)
  * aws-sdk (1.7.1)
  * backports (2.6.7)
  * binding_of_caller (0.6.7)
  * builder (3.0.4)
  * bundler (1.1.5)
  * choice (0.1.6)
  * coderay (1.0.7)
  * cube-ruby (0.0.2)
  * descendants_tracker (0.0.1)
  * diff-lcs (1.1.3)
  * erubis (2.7.0)
  * excon (0.15.4)
  * factory_girl (3.5.0)
  * factory_girl_rails (3.5.0)
  * faraday (0.8.4)
  * foreman (0.60.2)
  * friendly_id (4.0.7)
  * grape (0.4.0)
  * grape-entity (0.2.0)
  * hashie (2.0.3)
  * heroku (2.30.1)
  * heroku-api (0.3.1)
  * hike (1.2.1)
  * httparty (0.9.0)
  * i18n (0.6.4)
  * journey (1.0.4)
  * json (1.7.3)
  * kgio (2.7.4)
  * launchy (2.1.1)
  * mail (2.4.4)
  * method_source (0.8)
  * mime-types (1.19)
  * money (5.0.0)
  * mono_logger (1.0.0)
  * multi_json (1.7.2)
  * multi_xml (0.5.3)
  * multipart-post (1.1.5)
  * netrc (0.7.5)
  * nokogiri (1.5.5)
  * pg (0.14.0)
  * plymouth (0.3.2)
  * polyglot (0.3.3)
  * powder (0.1.8)
  * pry (0.9.10)
  * pry-exception_explorer (0.1.9)
  * pry-nav (0.2.2)
  * pry-stack_explorer (0.4.5)
  * rack (1.4.5)
  * rack-accept (0.4.5)
  * rack-cache (1.2)
  * rack-cors (0.2.7)
  * rack-mount (0.8.3)
  * rack-protection (1.5.0)
  * rack-ssl (1.3.2)
  * rack-test (0.6.1)
  * rails (3.2.6)
  * rails-erd (1.1.0)
  * railties (3.2.6)
  * raindrops (0.10.0)
  * rake (0.9.2.2)
  * rdoc (3.12)
  * redis (3.0.3)
  * redis-namespace (1.2.1)
  * resque (1.24.1)
  * resque-cleaner (0.2.10)
  * resque-meta (2.0.0)
  * resque-retry (1.0.0)
  * resque-scheduler (2.0.1)
  * rest-client (1.6.7)
  * rspec (2.11.0)
  * rspec-core (2.11.1)
  * rspec-expectations (2.11.1)
  * rspec-mocks (2.11.1)
  * rspec-rails (2.11.0)
  * ruby-graphviz (1.0.8)
  * rubyzip (0.9.9)
  * rufus-scheduler (2.0.18)
  * shoulda-matchers (1.2.0)
  * sinatra (1.3.6)
  * slop (3.3.2)
  * sprockets (2.1.3)
  * thor (0.15.4)
  * tilt (1.3.3)
  * timecop (0.3.5)
  * treetop (1.4.10)
  * tzinfo (0.3.33)
  * unicorn (4.5.0)
  * uuidtools (2.1.3)
  * vegas (0.1.11)
  * virtus (0.5.4)
  * yajl-ruby (1.1.0)
```
